source/features/conflict-marker.tsx

```diff
@@ -72,25 +51,22 @@ async function init(): Promise<false | void> {
 	}
 }
 
+function init(signal: AbortSignal): void {
+	observe('#js-issues-toolbar', addConflictMarkers, {signal});
```

Comment:
What is this element? Does it exist on the milestones page? Does it work there now?

Reply:
It works ;)
<img width="409" alt="image" src="https://github.com/refined-github/refined-github/assets/50487467/a52f4f70-4d49-42e2-b308-3867d3402511">

Reply:
I think this will fail on slower connections because the observer will call the callback as soon as `#js-issues-toolbar` is found, but at that point the long list of PRs might not be done downloading yet. This is why I was suggesting to use batched-function and the link selector

```js
const addIcon = batchedFunction((icons) => {
  // create query, etc
});

observe('.js-issue-row:has(.octicon-git-pull-request.color-fg-open) a.js-navigation-open', addIcon, {signal})
```

Reply:
🤔 I've tested a lot but It can detect icons successly. But I'll test more slow network conditions.
And is your suggestion works correctly? As I know, the selector-observe only works as single element, not all elements.

Reply:
> As I know, the selector-observe only works as single element, not all elements

No, the observer detects any and all elements that appear on the page until the next page load.

Here's what happens with the suggested code:

1. `observe` finds 3 elements
2. calls `addIcon` once per element
3. batched function then calls the anonymous function with a single array: `[prLink1, prLink2, prLink3]`
4. you can use this array to construct the GQL call
5. if more items are detected later, it repeats from step 1

On fast connections maybe batchedFunction will call the function with 25 items, on slower connections it'll make multiple API calls, that's ok

Reply:
Exactly, that was I thought.
I'm just wondering is there existing some utility function: batchedFunction, and the batchedFunction's throttle (or debounce) time.

Reply:
What do you mean? It does exist and we're already using it. We don't need to change it at this point https://github.com/Richienb/batched-function